app/src/main/java/com/github/se/eduverse/ui/timetable/TimeTableScreen.kt

```diff
@@ -0,0 +1,486 @@
+                                if (addedTodo == actualTodos[index]) {
+                                  Icon(Icons.Rounded.CheckCircle, "Checked")
+                                } else {
+                                  Icon(Icons.Rounded.RadioButtonUnchecked, "Unchecked")
+                                }
+                                Text(actualTodos[index].name)
+                              }
+                        }
+                  }
+                }
+                DateAndTimePickers(
+                    context,
+                    selectedDate,
+                    lengthHour,
+                    lengthMin,
+                    selectDate = { selectedDate = it },
+                    selectTime = { hour, min ->
+                      lengthHour = hour
+                      lengthMin = min
+                    })
+              }
+        } else {
+          Column(
+              modifier = Modifier.fillMaxWidth().testTag("addEventDialog"),
+              verticalArrangement = Arrangement.spacedBy(8.dp)) {
+                Text("Name", fontWeight = FontWeight.Bold, fontSize = 24.sp)
+                OutlinedTextField(
+                    value = name,
+                    modifier = Modifier.fillMaxWidth(0.9f).testTag("nameTextField"),
+                    onValueChange = { name = it },
+                    placeholder = { Text("Name the task") })
+                DateAndTimePickers(
+                    context,
+                    selectedDate,
+                    lengthHour,
+                    lengthMin,
+                    selectDate = { selectedDate = it },
+                    selectTime = { hour, min ->
+                      lengthHour = hour
+                      lengthMin = min
+                    })
+              }
+        }
+      },
+      confirmButton = {
+        TextButton(
+            onClick = {
+              onDismiss()
+              timeTableViewModel.addScheduled(
+                  Scheduled(
+                      timeTableViewModel.getNewUid(),
+                      newElementType,
+                      selectedDate,
+                      ((lengthHour + lengthMin.toDouble() / 60) * millisecInHour).toLong(),
+                      idTaskOrEvent,
+                      timeTableViewModel.auth.currentUser!!.uid,
+                      name))
+              idTaskOrEvent = ""
+              name = ""
+              selectedDate = Calendar.getInstance()
+              lengthHour = 1
+              lengthMin = 0
+              addedTodo = null
+            },
+            modifier = Modifier.testTag("confirm"),
+            enabled = newElementType == ScheduledType.EVENT || addedTodo != null) {
+              Text("Confirm")
+            }
+      },
+      dismissButton = {
+        TextButton(onClick = { onDismiss() }, modifier = Modifier.testTag("cancel")) {
+          Text("Cancel")
+        }
+      })
+}
+
```

Comment:
The DialogCreate composable contains duplicate code for event and task creation, which increases the code size and makes it harder to maintain. You could refactor the code to extract common parts into reusable functions or composables. For example, create a separate composable for the date and time pickers, or for displaying the title and name field.